libs/parallelism/lcos_local/include/hpx/lcos_local/channel.hpp

```diff
@@ -252,7 +252,14 @@ namespace hpx { namespace lcos { namespace local {
                 empty_ = false;
                 push_active_ = false;
             }
-            void set_deferred(T&& val) noexcept
+            void set_deferred(T&& val)
+            // CUDA versions less than 11.3 don't compile push_pt correctly if
+            // this is noexcept. hpx::util::result_of (and std::result_of)
+            // inside deferred_call does not detect that the call is valid, and
+            // compilation fails.
+#if !defined(HPX_CUDA_VERSION) || (HPX_CUDA_VERSION >= 1130)
```

Comment:
```suggestion
#if !defined(HPX_CUDA_VERSION) || (HPX_CUDA_VERSION >= 1103)
```
if the version check should indeed be changed.

Reply:
I checked on daint with cuda 11.3 and it's also failing so I'll keep the 1103 workaround instead of 1130

Reply:
Actually it fails on LSU with the workaround of 1103, I'll check more in details which version fail and which doesn't
https://cdash.cscs.ch/viewBuildError.php?buildid=172624

Reply:
The problem ended up being the version number handled as octal values, the problem is reproducible with cuda 11.4 so I extended the workaround for this version. The latest version being 11.4 I'll see if I can find any tickets about this